pandas/io/formats/format.py

```diff
@@ -565,315 +583,446 @@ def __init__(
+        assert self._col_space is not None
 
-        self.header = header
-        self.index = index
-        self.line_width = line_width
-        self.max_rows = max_rows
-        self.min_rows = min_rows
-        self.max_cols = max_cols
-        self.max_rows_displayed = min(max_rows or len(self.frame), len(self.frame))
-        self.show_dimensions = show_dimensions
-        self.table_id = table_id
-        self.render_links = render_links
+    @property
+    def max_rows_displayed(self) -> int:
+        return min(self.max_rows or len(self.frame), len(self.frame))
 
-        if justify is None:
-            self.justify = get_option("display.colheader_justify")
+    def _is_in_terminal(self) -> bool:
+        """Check if the output is to be shown in terminal."""
+        return bool(self.max_cols == 0 or self.max_rows == 0)
+
+    @property
+    def max_cols_adj(self) -> Optional[int]:
+        """Number of columns fitting the screen."""
+        self._max_cols_adj: Optional[int]
+
+        if hasattr(self, "_max_cols_adj"):
+            return self._max_cols_adj
+
+        if not self._is_in_terminal():
+            self._max_cols_adj = self.max_cols
+            return self._max_cols_adj
+
+        width, _ = get_terminal_size()
+        if self._is_screen_narrow(width):
+            self._max_cols_adj = width
         else:
-            self.justify = justify
+            self._max_cols_adj = self.max_cols
+        return self._max_cols_adj
 
-        self.bold_rows = bold_rows
-        self.escape = escape
+    def _is_screen_narrow(self, max_width) -> bool:
+        return bool(self.max_cols == 0 and len(self.frame.columns) > max_width)
 
-        if columns is not None:
-            self.columns = ensure_index(columns)
-            self.frame = self.frame[self.columns]
+    @property
+    def max_rows_adj(self) -> Optional[int]:
+        """Number of rows fitting the screen."""
+        if not self._is_in_terminal():
+            return self.max_rows
+
+        _, height = get_terminal_size()
+        if self.max_rows == 0:
+            # rows available to fill with actual data
+            return height - self._get_number_of_auxillary_rows()
+
+        max_rows: Optional[int]
+        if self._is_screen_short(height):
+            max_rows = height
         else:
-            self.columns = frame.columns
+            max_rows = self.max_rows
 
-        self._chk_truncate()
-        self.adj = get_adjustment()
+        if max_rows:
+            if (len(self.frame) > max_rows) and self.min_rows:
+                # if truncated, set max_rows showed to min_rows
+                max_rows = min(self.min_rows, max_rows)
+        return max_rows
 
-    def _chk_truncate(self) -> None:
-        """
-        Checks whether the frame should be truncated. If so, slices
-        the frame up.
-        """
-        from pandas.core.reshape.concat import concat
+    def _get_number_of_auxillary_rows(self) -> int:
```

Comment:
doc-string

Reply:
as its not obvious what auxillary is (reading down it becomes clear but a doc-string is good)

Reply:
Did that.